### PR TITLE
Feature instance archive pdf

### PR DESCRIPTION
**Pull request by singhsanket143**
The pull request is implementation for the feature listed in issue #16620. Backend of the archive instance functionality after pressing the corresponding button to initiate the process is has been implementead. After the process is complete a mail is sent to the admin of the instance

image
 
This is the format of the archive pdf sent
image

### CODE REVIEW

File: client/views/helpers/nav/nav.scss [hunk 1]
```diff
@@ -151,6 +151,18 @@ $screen-sm-max: ($screen-md - 1);
...
 
+#navArchive {
+	background-color: #4286f4;
+
+	&:hover {
+		background-color: #88adea;
+	}
+
+	.fa {
+		font-size: 1em !important;
```
Comment (peter-hank) [inline client/views/helpers/nav/nav.scss]: @singhsanket143 can you try to not to use `!important`? Generally there is always a better way to do it.

File: server/methods.js [hunk 2]
```diff
@@ -675,4 +677,83 @@ Meteor.methods({
...
+    let doc = new PDFDocument({size: 'A4', margin: 50});
+
+    if (!doc) {
+      console.log('Not able to initialize PDFKit object');
+      return false;
+    }
+
+    let currentInstance = Instances.findOne({slug: slug});
+    let randomNum = Math.floor(Math.random() * (10000 - 1 + 1)) + 1;
+    let filename = 'Instance_archive_'+currentInstance.tablename+'_'+randomNum+'.pdf';
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 can you add spaces next to `+`?

File: server/methods.js [hunk 3]
```diff
@@ -675,4 +677,83 @@ Meteor.methods({
...
+    let headingFontSize = 28;
+    let normalFontSize = 14;
+    let doc = new PDFDocument({size: 'A4', margin: 50});
+
+    if (!doc) {
+      console.log('Not able to initialize PDFKit object');
+      return false;
+    }
+
+    let currentInstance = Instances.findOne({slug: slug});
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 can you add spaces next to `{` and `}`?

File: server/methods.js [hunk 4]
```diff
@@ -675,4 +677,83 @@ Meteor.methods({
...
+              filename: filename,
+              path: filepath
+          }
+        ]
+      });
+    } catch(err) {
+      console.log('Unable to send email. Something went wrong');
+      console.log(err);
+    } finally {
+      // fs.unlink(filepath);
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 is it not needed?

File: server/methods.js [hunk 5]
```diff
@@ -675,4 +677,83 @@ Meteor.methods({
...
+    }
+    console.log("OK")
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 can you add `;` at the end of the line and use `'` instead of `"`?

File: server/methods.js [hunk 6]
```diff
@@ -675,4 +677,83 @@ Meteor.methods({
...
     return Accounts.createUser({ email, password, profile: { name: profileName } });
   },
+  createPDF(slug) {
+    let questionList = [];
+    let answerList = [];
+    let boldFont = 'Helvetica-Bold';
+    let normalFont = 'Helvetica';
+    let headingFontSize = 28;
+    let normalFontSize = 14;
+    let doc = new PDFDocument({size: 'A4', margin: 50});
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 can you add spaces next to { and }?

File: server/methods.js [hunk 7]
```diff
@@ -675,4 +677,83 @@ Meteor.methods({
 
     return Accounts.createUser({ email, password, profile: { name: profileName } });
   },
+  createPDF(slug) {
```
Comment (peter-hank) [inline server/methods.js:680]: @singhsanket143 can you make sure here if the current user is admin?

**Review by peter-hank — CHANGES_REQUESTED** [review]

File: server/methods.js (hunk 7 above)
Comment (singhsanket143) [reply server/methods.js:680]: @peter-hank It is already handled

File: client/views/helpers/nav/nav.js [hunk 8]
```diff
@@ -37,9 +37,14 @@ Template.nav.events({
   },
   'click #navCode': function(event, template) {
     const parentNode = document.getElementById('nav');
-    popoverTemplate = Blaze.renderWithData(Template.qr_code, {
+    popoverTemplate = Blaqze.renderWithData(Template.qr_code, {
```
Comment (peter-hank) [inline client/views/helpers/nav/nav.js]: @singhsanket143 can fix the typo?

File: server/methods.js (hunk 7 above)
Comment (peter-hank) [reply server/methods.js:680]: > @peter-hank It is already handled

@singhsanket143 where exactly?

**Review by peter-hank — CHANGES_REQUESTED** [review]

File: server/methods.js (hunk 7 above)
Comment (singhsanket143) [reply server/methods.js:680]: @peter-hank The archive button is not rendered for a non-admin user. this is handled on the frontend

File: client/views/helpers/nav/nav.js (hunk 8 above)
Comment (singhsanket143) [reply client/views/helpers/nav/nav.js]: @peter-hank Done

File: server/methods.js (hunk 7 above)
Comment (singhsanket143) [reply server/methods.js:680]: @peter-hank I have also added logic to the backend now

**Comment by singhsanket143** [comment]
I have added a loader kind of thing on the download archive button and also the popup notification along with that.  @peter-hank

File: server/methods.js [hunk 9]
```diff
@@ -675,4 +677,87 @@ Meteor.methods({
...
+    // Draw the bounding borders
+    doc.rect(5, 5, 580, doc.y).stroke();
+
+    doc.writeSync(filepath);
+    this.unblock();
+    // Send email to admin with file as attachment
+    try {
+      Email.send({
+        to: currentInstance.admin,
+        from: currentInstance.admin,
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 can you replace it with `MAIL_ID_FROM`? The sender here should be the main app email address.

**Review by peter-hank — CHANGES_REQUESTED** [review]

File: server/methods.js [hunk 10]
```diff
@@ -675,4 +677,87 @@ Meteor.methods({
...
+        subject: 'Subject',
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 can you make the subject and the text more useful?

File: server/methods.js (hunk 9 above)
Comment (singhsanket143) [reply server/methods.js]: @peter-hank Updated the from value

File: server/methods.js (hunk 10 above)
Comment (singhsanket143) [reply server/methods.js]: @peter-hank Updated the text and subject

File: server/methods.js [hunk 11]
```diff
@@ -675,4 +677,87 @@ Meteor.methods({
...
+    doc.moveDown();
+    let questionsOnCurrentInstance = Questions.find({ instanceid: currentInstance._id });
+    let noOfQuestionsOnCurrentInstance = Questions.find({ instanceid: currentInstance._id }).count();
+    let answersOnCurrentInstance = Answers.find({ instanceid: currentInstance._id });
+    answersOnCurrentInstance.forEach(function(ans) {
+      answerList.push(ans);
+    });
+    doc.fontSize(normalFontSize);
+    doc.text(`Author: ${currentInstance.author}`, 20, doc.y, {align: 'left'});
+    doc.text(`Description of instance: ${currentInstance.description}`, 20, doc.y, {align: 'left'});
```
Comment (peter-hank) [inline server/methods.js]: @singhsanket143 

can you replace?

`Description of instance` > `Description Of The Instance`
`Max Question Allowed` > `Max. Questions Allowed`
`Max Response Allowed` > `Max. Responses Allowed`
`Total Questions Posted Till Now` > `Total Questions Posted Until Now`
`List of replies` > `List Of Replies`

**Review by peter-hank — CHANGES_REQUESTED** [review]
Just some typos.

File: server/methods.js (hunk 11 above)
Comment (singhsanket143) [reply server/methods.js]: @peter-hank Updated the file

**Pull request merged**